devtools/plugins/desktop/common/src/communication-layer/index.ts

```diff
@@ -0,0 +1,109 @@
+import type {
+  ExtensionSupportedEvents,
+  MessengerOptions,
+} from "@player-tools/devtools-types";
+import { useEffect, useState } from "react";
+
+type IntoArrays<T> = {
+  [P in keyof T]: T[P][];
+};
+
+type CommunicationLayerMethods = Pick<
+  MessengerOptions<ExtensionSupportedEvents>,
+  "sendMessage" | "addListener" | "removeListener"
+>;
+
+type Callbacks = IntoArrays<CommunicationLayerMethods>;
+
+/** Web extension communication layer leverage by the @player-tools/devtools-messenger */
+export const useCommunicationLayer = (): Pick<
+  MessengerOptions<ExtensionSupportedEvents>,
+  "sendMessage" | "addListener" | "removeListener"
+> => {
+  const [layerCallbacks, setLayerCallbacks] = useState<Callbacks>({
+    sendMessage: [],
+    addListener: [],
+    removeListener: [],
+  });
+
+  useEffect(() => {
+    // TODO:: figure out the connection errors when uncommenting this block
```

Comment:
is there an associated issue for this `TODO`?

Reply:
https://github.com/player-ui/tools/issues/95